trail/trail.go

```diff
@@ -82,6 +84,11 @@ func ToGRPC(err error) error {
 	if err == nil {
 		return nil
 	}
+
+	if errors.Is(err, io.EOF) {
```

Comment:
Could you add some test coverage for this?

Maybe just expand https://github.com/gravitational/trace/blob/master/trail/trail_test.go#L94 to include `io.EOF`